# My-repo-SLE777
ASSIGNMENT 4 (Part 1)- Data Wrangling, Analysis and Plotting

# Overview
This repository has R tools and a R Markdown report that can be used with RStudio and GitHub to organize data, display it, and analyze sequence variation.  Parts 1 and 2 are both covered. Part 1 has points 1 through 10; Part 2 has points 1 through 6.  In Part 1, the tools bring in and look at two sets of data (gene_expression.tsv and growth_data.csv), make plots, create descriptive statistics, and test hypotheses.  In Part 2, they get coding DNA sequences (CDS) for E. coli and Saprospirales bacteria and then figure out data at the sequence level, like gene counts, nucleotide rates, codon usage bias, and k-mer enrichment.  All of the code works perfectly, produces results that can be repeated, and can be put together into a single, complete report using R Markdown.

# Repository Structure
The repository is set up so that it is easy to understand and use again.  Files like gene_expression.tsv and growth_data.csv are stored in the data/ folder.  There are two subfolders, part1/ and part2/, inside the outputs/ folder that hold all the figures and tables that were made.  There are several R scripts in the scripts/ directory. Each one does a different part of the project, from importing and organizing data to visualizing it and analyzing sequences.  These tools are put together in the main R Markdown file, report.Rmd, to make the end report.  For faster work in RStudio and GitHub, supporting files like.Rproj and.gitignore are included.

# Software Requirments
